src/solvers/prop/prop_conv.h

```diff
@@ -10,18 +10,16 @@ Author: Daniel Kroening, kroening@kroening.com
 #ifndef CPROVER_SOLVERS_PROP_PROP_CONV_H
 #define CPROVER_SOLVERS_PROP_PROP_CONV_H
 
-#include "decision_procedure.h"
+#include "prop_decision_procedure.h"
 
 // API that provides a "handle" in the form of a literalt
 // for expressions.
 
-class prop_convt:public decision_proceduret
+class prop_convt : public prop_decision_proceduret
```

Comment:
That means `smt2_convt` still implements `prop_decision_proceduret`.

The more I think about this PR the more doubts I have.

The predicate handle interface is something useful to have also in `smt2`. It is required for some of our algorithms and hence it doesn't make sense not to have a common interface for SAT-based implementations and SMT-based ones. IMO it should be generalised, renamed and moved up into `solvers/`.